node/src/integration-test/kotlin/net/corda/node/services/rpc/RpcReconnectTests.kt

```
@@ -0,0 +1,543 @@
+                        .states
+
+                assertTrue(allCashStates.size >= flowProgressEvents!!.size, "Not all flows were executed successfully")
+
+                // Check that no flow was triggered twice.
+                val duplicates = allCashStates.groupBy { it.state.data.amount }.filterValues { it.size > 1 }
+                assertTrue(duplicates.isEmpty(), "${duplicates.size} flows were retried illegally.")
+            }
+        }
+    }
+
+    /**
+     * Simple proxy that can be restarted and introduces random latencies.
+     */
+    class TestLocalProxy(val serverPort: Int, val remotePort: Int) {
+
+        private var serverSocket: ServerSocket? = null
+
+        private val stopCopy = AtomicBoolean(false)
+
+        fun start(): TestLocalProxy {
+            stopCopy.set(false)
+            serverSocket = ServerSocket(serverPort)
+            thread {
+                try {
+                    while (!serverSocket!!.isClosed) {
+                        val socket = serverSocket!!.accept()
+                        thread {
+                            try {
+                                val target = Socket("localhost", remotePort)
+                                // send message to node
+                                thread {
+                                    try {
+                                        socket.getInputStream().slowCopyTo(target.getOutputStream())
+                                    } catch (e: Exception) {
+                                    }
+                                }
+                                target.getInputStream().slowCopyTo(socket.getOutputStream())
+                            } catch (e: Exception) {
+                                // the target is dead
+                            } finally {
+                                try {
+                                    socket.close()
+                                } catch (e: Exception) {
```

I'm a bit weary of all these ignore exception blocks. Is there a reason why we want to ignore the exception on close? Either add comments in all the places where exceptions are ignore, or revert to the use of `use`. I would much prefer the later.